File: python/basics/classes.py
class Money (object):
    def __init__(self, amount, currency):
        self.amount = amount
        self.currency = currency

    def __str__(self):
        return str(self.amount) + ' ' + self.currency

# create a new instance of class
money = Money(220, 'EUR')
money.amount, money.currency
# (220, 'EUR')

print(money)
# 220 EUR

# subclasses
class VirtualMoney (Money):
    def __init__(self, date):
        self.date = date

    def __str__(self):
        return str(self.amount) + ' ' + self.currency + ' (use before ' + self.date + ')'

virtual_money = VirtualMoney('2018-12-31')
virtual_money.amount = 20
virtual_money.currency = 'DIS'

print(virtual_money)
# 20 DIS (use before 2018-12-31)

File: _Scala/Primer/classes.scala
class Money(val a:Int, val c:String) {
    var amount: Int = a
    var currency: String = c
    
    // override
    override def toString = amount.toString ++ " " ++ currency
}

// create instance of class
val money = new Money(220, "EUR")
money.amount, money.currency                // (220,EUR)

println(money)                              // 220 EUR

// subclasses
class VirtualMoney(override val a:Int, override val c:String, val d:String) extends Money(a, c) {
    var date: String = d
    
    // override
    override def toString = amount.toString ++ " " ++ currency ++ " (use before " ++ date ++ ")"
}

val virtual_money = new VirtualMoney(20, "DIS", "2018-12-31")

println(virtual_money)                      // 20 DIS (use before 2018-12-31)